core-httpclient-impl/src/main/java/com/optimizely/ab/HttpClientUtils.java

```diff
@@ -39,13 +39,11 @@ private HttpClientUtils() {
         .build();
 
     public static RequestConfig getDefaultRequestConfigWithTimeout(int timeoutMillis) {
-        if (requestConfigWithTimeout == null) {
```

Comment:
I understand we try to re-use RequestConfig once created. Curious why we do not want to reuse?

Reply:
this function is getting used at multiple places (for setting request config with timeout of SegmentAPIManager as well as for EventAPIManager). If we want to use different timeout for different Request config then this check will not allow that as it will return the first object that was initialized.